packages/server/lib/controllers/appAuth.controller.ts

```diff
@@ -179,24 +182,38 @@ class AppAuthController {
                 environmentId: environment.id,
                 accountId: account.id
             });
+            if (!updatedConnection) {
+                await logCtx.error('Failed to create connection');
+                await logCtx.failed();
+                return publisher.notifyErr(res, wsClientId, providerConfigKey, connectionId, WSErrBuilder.UnknownError('failed to create connection'));
+            }
 
-            if (updatedConnection) {
-                await logCtx.enrichOperation({ connectionId: updatedConnection.connection.id!, connectionName: updatedConnection.connection.connection_id });
-                void connectionCreatedHook(
-                    {
-                        connection: updatedConnection.connection,
-                        environment,
-                        account,
-                        auth_mode: 'APP',
-                        operation: updatedConnection.operation
-                    },
-                    session.provider,
-                    logContextGetter,
-                    undefined,
-                    logCtx
-                );
+            if (session.connectSessionId) {
+                const connectSession = await getConnectSession(db.knex, { id: session.connectSessionId, accountId: account.id, environmentId: environment.id });
+                if (connectSession.isErr()) {
+                    await logCtx.error('Failed to get session');
+                    await logCtx.failed();
+                    return publisher.notifyErr(res, wsClientId, providerConfigKey, connectionId, WSErrBuilder.UnknownError('failed to get session'));
```

Comment:
should we failed the connection creation in this case? we'll end up returning a failure even though the connection db upsert above was successful

Reply:
ideally we'll add a transaction that's why I kept the `db` params, I feel like it's a very unlikely event for the moment. 
I'll create a followup issue if that's alright?

Reply:
sounds reasonable